project2/get_ingredients.py

```diff
@@ -0,0 +1,112 @@
+# input "1 cup white sugar"
+# output
+    # ingredient: sugar
+    # quantity: 1
+    # measurement: cup
+    # descriptior: white
+    # preparation: ''
+
+# input "1 cup butter, softened"
+# output
+    # ingredient: butter
+    # quantity: 1
+    # measurement: cup
+    # descriptior: ''
+    # preparation: 'soften'
+
+import re
+import nltk
+from nltk.corpus import stopwords
+from collections import Counter
+
+def clean_ingredients(ing):
+    # basic cleaning
+        # remove all punctiation
+        # keep / : 1/5 pounds
+        # keep . : 1.5 pounds
+    pattern = '[,;!$%^&*()]'
+    ing = ' '.join(re.sub(pattern," ",ing).split())
+    return ing
+
+
+def normalize_ingredients(ing):
+    ing = clean_ingredients(ing)
+    # convert all words to lower case
+    ing_lower = ing.lower()
+    # tokenize all tweet text
+    ing_tokens = nltk.word_tokenize(ing_lower)
+
+    # default nltk stop words
+    stop_words_nltk = stopwords.words('english')
+    # remove English stop words based on default nltk package
+    ing_normalized_text = [word for word in ing_tokens if word not in stop_words_nltk]
+
+    return ing_normalized_text
+
+
+def get_phrases_from_chunks(chunk_gram, ing_pos):
+    chunkParser = nltk.RegexpParser(chunk_gram)
+    chunked = chunkParser.parse(ing_pos)
+    matches = []
+    for subtree in chunked.subtrees(filter=lambda t: t.label() == 'Chunk'):
+        leaves_words = [ leaf[0] for leaf in subtree.leaves() ]
+        phrase = ' '.join(leaves_words)
+        matches.append(phrase)
+
+    return matches
+
+
+def get_ingredients(ing_string):
```

Comment:
this is the main function of the module. It is taking in a string and returning our internal representation of an ingredient. `get_ingredients` is kind of confusing, partially because it is plural for some reason. `parse_ingredient` might be a more semantically accurate name for what this function is doing

Reply:
It would also be nice if this module supplied a `parse_ingredients` function that mapped `parse_ingredient` over a list of strings and returned the result.